### ISSUE TITLE
Running applyrules.py and runrulemine.py

### ISSUE THREAD

**Issue by vishrantsaagar** [HKUST-KnowComp/RINANTE#4]
Hey, I was actually trying to understand your paper and code because I'm currently in an AI class and it was quite interesting to me to analyze how you went through this process. I am currently stuck as it appears that the only times that running applyrules.py and runrulemine.py works is when I use se14l to find aspect terms, but the required data for the other files seems to be missing. How do you generate the data so that we can get the aspect and opinion terms for all the datasets involved?

**Comment by hldai** [comment]
Hi, you need to get the corresponding dependency parsing and POS tagging results. I used Stanford NLP.

**Comment by vishrantsaagar** [comment]
Oh, thanks a lot for the prompt reply. So in your config.py file you have the following code: 
```
SE14L_FILES = {
    'train_sents_file': join(SE14_DIR, 'laptops/laptops_train_sents.json'),
    'train_tok_texts_file': join(SE14_DIR, 'laptops/laptops_train_texts_tok_pos.txt'),
    'train_valid_split_file': join(SE14_DIR, 'laptops/laptops_train_valid_split.txt'),
    'test_sents_file': join(SE14_DIR, 'laptops/laptops_test_sents.json'),
    'test_tok_texts_file': join(SE14_DIR, 'laptops/laptops_test_texts_tok_pos.txt'),
    'train_dep_tags_file': join(SE14_DIR, 'laptops/laptops-train-rule-dep.txt'),
    'train_pos_tags_file': join(SE14_DIR, 'laptops/laptops-train-rule-pos.txt'),
    'word_cnts_file': join(SE14_DIR, 'laptops/word_cnts.txt'),
    'aspect_rule_patterns_file': join(DATA_DIR, 'rule_patterns/se14l_aspect_mined_rule_patterns.txt'),
    'aspect_term_filter_file': join(SE14_DIR, 'laptops/se14l_aspect_term_filter_vocab.txt'),
    'aspect_term_hit_rate_file': join(SE14_DIR, 'laptops/se14l_aspect_term_hit_rate_file.txt')
} 
```
 About half of these files are already present in the GOOGLE DRIVE dataset given but the other half aren't. Are you saying that you used the Stanford CoreNLP **separately** to generate those files and dragged and dropped them into the folder for use?

**Comment by vishrantsaagar** [comment]
Also, why does it not work for opinion mining for se14l despite of having the dependency parsing and POS tagging results included? It seems to be that the files "se14l_opnion_term_filter_vocab.txt" and "se14l_opinion_term_hit_rate_file.txt". How do you generate those?

**Comment by vishrantsaagar** [comment]
If you still have the code for how you generated the dep_tags, pos_tags and term_filter_vocab.txt and term_hit_rate_file.txt, please?

**Comment by vishrantsaagar** [comment]
Could you possibly make the code available to obtain the  corresponding dependency parsing and POS tagging results? Also beyond that, after the term_filer_file and hit_rate_file are generated, it requires certain other files which I believe may not be available,

**Comment by hldai** [comment]
Hi, I see your problems. But since I am working for a company now and don't have much time on weekdays, I will see what I can do this weekend.

**Comment by vishrantsaagar** [comment]
Oh yes, that would be great, thank you so much! Hope your work is going good as well!

**Comment by vishrantsaagar** [comment]
Hey, just wanted to send this as a reminder. Don't want to seem demanding. Just wanted to make sure. Really appreciate it.

**Comment by hldai** [comment]
I've update the java code for generating dependency parsing, pos tagging, and tokenization results.

**Comment by vishrantsaagar** [comment]
Hey, yeah, thank you so much. I've been messing around with it a little but just can't get it to run.
For 1, could you let me know if I have the order of execution correct:
Tokenize.java
depParse.java
runrulemine.py
applyrules.py
trainrinante.py
and then repeat for each dataset available?

If the above is correct. I am stuck on figuring out how to get Stanford NLP to work on the java files given, did you use any arguments? what command did you use on terminal to run the 2 java files to generate the desired output?

**Comment by hldai** [comment]
This repo already contains the tokenized semeval data, so you don't need to run Tokenize.java. You run DepParse.java to get dependency parsing and pos tagging results, runrulemine.py to mine the rules.
For the java code, I used StanforNLP 4.1.0. Before running it, you should change the file paths in the main function and the path to the model file. No arguments needed. 
If you cannot run the java code. Try google the problem or post the error here.

You run applyrules.py to apply the mined rules, but you probably need to get the Yelp data, or other restaurant review data yourself, since it seems I am not allowed to distribute Yelp data.

**Comment by vishrantsaagar** [comment]
Oh alright, I understand, I really appreciate the help so far.

**Comment by vishrantsaagar** [comment]
Hey, I was setting all of this up on IntelliJ because VSCode seemed hopeless with StanfordNLP

The following is my pom.xml file:
![image](https://user-images.githubusercontent.com/71482556/141015265-33f50c57-c484-412c-a8f6-c3e3b3d9afae.png)

I have the DepParse file in the same project
![image](https://user-images.githubusercontent.com/71482556/141015360-4b53735f-c3f5-432e-bd80-a95d6041e763.png)

I am running into this error:
![image](https://user-images.githubusercontent.com/71482556/141015417-ed2aeec4-93ca-4bb2-bfab-070a813e288b.png)


It is confusing because I included the models in my pom.xml file.


Do you have any idea what might be going wrong?
![image](https://user-images.githubusercontent.com/71482556/141015831-419a222f-8ed9-449e-86e0-235e81a6f9d0.png)

The error point is at line 96

**Comment by hldai** [comment]
Try setting taggerPath to the path of the tagger model file.

**Comment by vishrantsaagar** [comment]
Where could I find the tagger model file? I downloaded the following folder  "stanford-corenlp-4.3.1" from the site and it doesn't seem to have it

**Comment by hldai** [comment]
It should be in stanford-corenlp-4.3.1-models.jar, jar files are actually zip files. Alternatively, you can try adding this file to class path with -cp when you run the java code.

**Comment by vishrantsaagar** [comment]
do i need to have the  stanford-corenlp-4.3.1-models.jar in the same folder as well? which in this case would be com.vishrant.nlp? because even after moving it there and setting cp using command "javac -cp stanford-corenlp-4.3.1-models.jar DepParse.java" I still get these errors:

![image](https://user-images.githubusercontent.com/71482556/141231036-2fc04b03-85cf-446d-8f7e-dd0fb4623aeb.png)

and more saying the packages doesn't exist.


I even tried to just run the file by making the taggerPath this:
![image](https://user-images.githubusercontent.com/71482556/141231441-aa4bcca1-1b48-43d3-a01d-b36949a10902.png)

I directly used the "copy address text" feature to get the right folder and stuff. And then I get the following error:

![image](https://user-images.githubusercontent.com/71482556/141231539-d4566562-dfd5-4efe-81b0-d8f9cf7a1020.png)


Really trying it all here

**Comment by vishrantsaagar** [comment]
Hey, you can disregard the prior messages, but is there any particular reason you picked 150 epochs? did it just give you the most valid f1 score?

**Comment by vishrantsaagar** [comment]
And the final question:
You mention shared, double alt and pre configurations for RINANTE,
in the rinante.py code you have 2 places that mention:
    lstm_l2_src = False
    lstm_l2_tar = False

Can I get context on which is which? Thank you so much for all your help!

**Comment by hldai** [comment]
150 epochs: no particular reason, just so that a good f1 score on the validation set can be obtained.

lstm_l2_src and lstm_l2_tar are about whether or not to add l2 regularization to the LSTM parapemters. I believe they are always set to False so you can ignore them. share_lstm is the argument that controls shared and double.

**Comment by vishrantsaagar** [comment]
Oh then how exactly do you get the RINANTE alt and pre configurations?
